### ISSUE TITLE
False negative for `Lint/Void` for expressions inside brackets

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#12716]
Lint/Void leaves expressions like `(1 + 1)` alone because they are wrapped in brackets.
I tried fixing this myself but it is too complex for me.

## Steps to reproduce the problem
Testcase:
```rb
it "registers an offense for void op #{op} if used in brackets" do
  expect_offense(<<~RUBY, op: op)
    (a %{op} b)
       ^{op} Operator `#{op}` used in void context.
    ((a %{op} b))
        ^{op} Operator `#{op}` used in void context.
    (((a %{op} b)))
  RUBY

  # Rest handled by Style/RedundantParentheses and Layout/IndentationConsistency
  # Though it would look better here if brackets are removed directly
  expect_correction(<<~RUBY)
    (a
    b)
    ((a
    b))
    (((a %{op} b))) # or maybe just `à %{op} b`? Either way, shouldn't be autocorrected like the others
  RUBY
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.60.2 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-performance 1.18.0
  - rubocop-rspec 2.23.2
```
